eng/common/native/init-compiler.sh

```diff
@@ -112,7 +112,7 @@ if [[ -z "$CC" ]]; then
 fi
 
 if [[ "$compiler" == "clang" ]]; then
-    if command -v "lld$desired_version" > /dev/null; then
+    if command -v lld || command -v "lld$desired_version" > /dev/null; then
```

Comment:
nit: perhaps would be better if we parse version specific variant before version agnostic one (in case both are available)?
```suggestion
    if command -v "lld$desired_version" || command -v lld > /dev/null; then
```